src/GitHub/PullRequests.hs

```diff
@@ -42,20 +48,37 @@ pullRequestsToAst PullRequests{..} = QueryNode
 
 {- | Arguments for the 'PullRequest' connection.
 -}
-data PullRequestsArgs = PullRequestsArgs
-    { pullRequestsLast   :: !Int
-    , pullRequestsStates :: !(NonEmpty State)
+data PullRequestsArgs (args :: [ArgsType]) = PullRequestsArgs
+    { pullRequestsArgsLast   :: !Int
+    , pullRequestsArgsStates :: !(NonEmpty State)
+    }
+
+instance HasLimit PullRequestsArgs where
+    lastL = lens pullRequestsArgsLast (\args new -> args { pullRequestsArgsLast = new })
+    {-# INLINE lastL #-}
+
+instance HasStates PullRequestsArgs where
+    statesL = lens pullRequestsArgsStates (\args new -> args { pullRequestsArgsStates = new })
+    {-# INLINE statesL #-}
+
+{- | Default value of 'PullRequestsArgs'. Use methods of 'HasLimit' and
+'HasStates' to change its fields.
+-}
+defPullRequestsArgs :: PullRequestsArgs '[ 'ArgsLimit, 'ArgsStates ]
```

Comment:
Maybe we need to provide type aliases, to ease users' interface? Or even we should think if we can leave this type opened to modifications anywhere?